tools/lint/sources.swiftlint.yml

```diff
@@ -70,6 +68,8 @@ whitelist_rules: # we enable lint rules explicitly - only the ones listed below
   - vertical_whitespace_opening_braces
   - void_return
   - xct_specific_matcher
+  # - missing_docs # TODO: RUMM-126 enforce documentation for public APIs
+  # - attributes
```

Comment:
Why disabling this rule? It ensures the attributes format. If disabled, all following will lint, so must be code reviewed:
```swift
// The way we want:
@available(iOS 12, *)
internal class Foo {}

// The way we don't want:
@available(iOS 12, *)

internal class Foo {}

// The way we don't want:
@available(iOS 12, *) internal class Foo {}
```

Reply:
at first i disabled it becuase it doesn't allow `@objc func foo() { ... }` on the same line, otherwise it looks bad in for example `RUM+Objc.swift` where many tiny methods are stacked.

then it also comes handy for the custom rule regex, i couldn't come up with a regex which works with `@objc` on the line above. if you can find one, we can re-enable this rule 👌 

i didn't know we didn't want `@available(iOS 12, *) internal class Foo {}`. it doesn't work or is it just too long?

Reply:
Our current code style enforces the rule on attributes. We shouldn't just disable it because we want to add another custom rule that conflicts with it. We should resolve this conflict instead - this means to include newline separator in the regexp.

Reply:
i can't get it working by adding a newline, can you please have a look?
https://regex101.com/r/1VlTj4/3

`^\s*(?!@objc)\n(override )?public` -> matches nothing
`^\s*(?!@objc\n)(override )?public` -> matches the wrong case below
```swift
@objc
override public foo()
```
---
with `attributes` re-enabled,
```swift
@objc public static func granted() -> DDTrackingConsent { .init(sdkConsent: .granted) }
@objc public static func notGranted() -> DDTrackingConsent { .init(sdkConsent: .notGranted) }
@objc public static func pending() -> DDTrackingConsent { .init(sdkConsent: .pending) }
```
will be
```swift
@objc
public static func granted() -> DDTrackingConsent { .init(sdkConsent: .granted) }

@objc
public static func notGranted() -> DDTrackingConsent { .init(sdkConsent: .notGranted) }

@objc 
public static func pending() -> DDTrackingConsent { .init(sdkConsent: .pending) }
```
do you prefer the latter?

Reply:
> do you prefer the latter?

It's not about my preference. Currently, we lint our code using `attributes` rule and I see no reason to disable it. 

This is valid syntax and it follows our linter:
```swift
@objc
public static func granted() -> DDTrackingConsent { .init(sdkConsent: .granted) }
```

I've been playing with the regex but cannot came up with the solution quickly. I can try later.

Reply:
i removed linter rule to unblock this PR